satpy/composites/__init__.py

```diff
@@ -334,7 +337,17 @@ def check_areas(self, data_arrays):
                       "'{}'".format(self.attrs['name']))
             raise IncompatibleAreas("Areas are different")
 
-        return data_arrays
+        # Drop neglible non-dimensional coordinates (understood to be "close-enough" for most operations)
```

Comment:
Could this be put in a separate function ? This looks like it adds a new functionality to the `check_areas` function, making it accomplish 2 tasks (as hinted by the docstring).

Reply:
The problem with removing this from `check_areas` is that all composites would then have to call `check_areas` *and* this new method. I'm ok if `check_areas` also does this operation for now although moving it to a separate function wouldn't be a bad idea as long as `check_areas` calls it.

Reply:
Ok, I looked more at this and I believe we need a little refactoring. What we really to in `check_areas` is match the different projectables together. For that we are now checking that the areas match, and we want to add dropping non-dimensional coordinates. Hence I would suggest having as main function something called eg `match_dataarrays`, that in turn calls `check_areas`, and calls a new function called eg `drop_non_dimensional_coordinates`.